kratos/utilities/parallel_utilities.h

```diff
@@ -72,6 +72,15 @@ class KRATOS_API(KRATOS_CORE) ParallelUtilities
 
     ///@}
 
+protected:
+
+    /** @brief Initializes members.
+     * Note that this method must only be called once at startup
+     * The Kernel takes care of it
+     */
+    friend class Kernel; // forward declaring to avoid the include ("parallel_utilities.h" is included in many places but "kernel.h" should not be!)
```

Comment:
Why do you need the forward declaration? Is it because you want to force it protected?

since `parallel_utilities.h` is already in the kernel, couldn't be easier just to move `static int * mspNumThreads` to the kernel (like `static bool mIsDistributedRun`) and then call the initialize directly with the kernel variable?:

`
if (!IsImported("KratosMultiphysics")) {
    this->ImportApplication(mpKratosCoreApplication);
    ParallelUtilities::Initialize(mspNumThreads);
}

PrintParallelismSupportInfo();
`

and then:

```
void ParallelUtilities::Initialize(int * rNumberOfThreads)
{
    static int num_threads;
    num_threads = InitializeNumberOfThreads();
    rNumberOfThreads = &num_threads;
}
```

This way you can declare `static int * mspNumThreads` as private and add a Getter in the kernel so parallel_utilities can retrieve the value.

the idea its a little bit different from what we have right now, but personally I think it make sense, since this way the kernel owns both the distributed and parallel info.